main/modes/utilities/ch32v003test/mode_ch32v003test.c

```diff
@@ -0,0 +1,144 @@
+//==============================================================================
+// Includes
+//==============================================================================
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "swadge2024.h"
+#include "mode_ch32v003test.h"
+#include "mainMenu.h"
+#include "hdw-ch32v003.h"
+
+//==============================================================================
+// Functions Prototypes
+//==============================================================================
+
+void ch32v003testEnterMode(void);
+void ch32v003testExitMode(void);
+void ch32v003testMainLoop(int64_t elapsedUs);
+int uprintf(const char* fmt, ...);
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+typedef struct
+{
+    font_t* font;
+    int64_t tElapsedUs;
+} ch32v003test_t;
+
+ch32v003test_t* ch32v003test;
+
+const char ch32v003testName[] = "Ch32v003test";
+
+swadgeMode_t modeCh32v003test = {
+    .modeName                 = ch32v003testName,
+    .wifiMode                 = NO_WIFI,
+    .overrideUsb              = false,
+    .usesAccelerometer        = false,
+    .usesThermometer          = false,
+    .overrideSelectBtn        = false,
+    .fnEnterMode              = ch32v003testEnterMode,
+    .fnExitMode               = ch32v003testExitMode,
+    .fnMainLoop               = ch32v003testMainLoop,
+    .fnAudioCallback          = NULL,
+    .fnBackgroundDrawCallback = NULL,
+    .fnEspNowRecvCb           = NULL,
+    .fnEspNowSendCb           = NULL,
+    .fnAdvancedUSB            = NULL,
+};
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+/**
+ * Enter the ch32v003test mode, allocate and initialize memory
+ */
+void ch32v003testEnterMode(void)
+{
+    // Allocate memory for this mode
+    ch32v003test = (ch32v003test_t*)heap_caps_calloc(1, sizeof(ch32v003test_t), MALLOC_CAP_8BIT);
+
+    // Load a font
+    font_t* ch32v003testFont = (font_t*)heap_caps_calloc(1, sizeof(font_t), MALLOC_CAP_8BIT);
+    loadFont(SONIC_FONT, ch32v003testFont, false);
+
+    // Load some fonts
+    ch32v003test->font = ch32v003testFont;
+
+    ch32v003test->tElapsedUs = 0;
+
+    size_t sz;
+    const uint8_t* buf = cnfsGetFile(LEDARRAY_CFUN_BIN, &sz);

```

Comment:
If this is the default file we'll probably want to move it to `app_main()`

Reply:
Agree.

Reply:
it was btw